app/V1Module/presenters/GroupsPresenter.php

```diff
@@ -337,6 +337,45 @@ public function actionSetArchived(string $id) {
     $this->sendSuccessResponse($this->groupViewFactory->getGroup($group));
   }
 
+  public function checkRelocate(string $id, string $newParentId) {
+    $group = $this->groups->findOrThrow($id);
+    $newParent = $this->groups->findOrThrow($newParentId);
+
+    if (!$this->groupAcl->canRelocate($group)
+      || !$this->groupAcl->canAddSubgroup($newParent)
+      || $group->isArchived()
+      || $newParent->isArchived()
+    ) {
+      throw new ForbiddenRequestException();
+    }
+  }
+
+  /**
+   * Relocate the group under a different parent.
+   * @POST
+   * @param string $id An identifier of the relocated group
+   * @param string $newParentId An identifier of the new parent group
+   * @throws NotFoundException
+   */
+  public function actionRelocate(string $id, string $newParentId) {
+    $group = $this->groups->findOrThrow($id);
+    $newParent = $this->groups->findOrThrow($newParentId);
+    
+    if ($group->getInstance() !== null && $group->getInstance()->getRootGroup() === $group) {
+      throw new ForbiddenRequestException("Group '$id' is the root group of instance '{$group->getInstance()->getId()}' and root groups cannot be relocated.");
+    }
+
+    foreach($newParent->getParentGroupsIds() as $parentId) {
+      if ($parentId === $id) { // group cannot be relocated under its descendant
+        throw new ForbiddenRequestException("Unable to relocate group '$id' under parent group '$newParentId' since it would create an inheritance loop.");
```

Comment:
Please add frontend error code and params... this looks like something that could be encountered in web-app